robottelo/ui/role.py

```diff
@@ -83,22 +83,48 @@ def get_resources(self, role_name):
         :param role_name: String with role name.
         :return: List of strings with resource names.
         """
-        self.search_and_click(role_name)
-        self.click(tab_locators['roles.tab_filters'])
+        self.search(role_name)
+        self.click(locators['roles.filters_button'] % (role_name, role_name))
+        # make sure The role filter page is loaded
+        self.wait_until_element(locators['role_filters.title'])
         resources = [
             resource.text for resource in
-            self.find_elements(locators['roles.resources'])
+            self.find_elements(locators['role_filters.resources'])
         ]
-        next_ = self.find_element(locators["roles.filters.pagination_next"])
+        next_ = self.find_element(locators['role_filters.pagination_next'])
         while next_:
             self.click(next_)
-            next_ = self.find_element(
-                locators["roles.filters.pagination_next"])
+            self.wait_until_element(locators["role_filters.results_ready"])
             resources.extend(
                 [resource.text for resource in
-                    self.find_elements(locators['roles.resources'])])
+                 self.find_elements(locators['role_filters.resources'])])
+            next_ = self.find_element(locators['role_filters.pagination_next'])
         return resources
 
+    def _get_page_resources_permissions(self, resource_types, permissions):
+        """Retrieve the permissions for each resource type in resource_types
+        on Role Filters page.
+
+         :param resource_types: the list of resource types to check on the
+            current page.
+        :param  permissions: a dict store to register the resource permissions
+            found.
+
+        Note: The resource is removed from resource_types when found
+        """
+        self.wait_until_element(locators["role_filters.results_ready"])
+        resources = [
+            resource.text for resource in
+            self.find_elements(locators['role_filters.resources'])
+            ]
+        for res_type in resources:
+            if res_type in resource_types:
+                perm_element = self.find_element(
+                    locators['role_filters.permissions'] % res_type)
+                if perm_element:
+                    permissions[res_type] = perm_element.text.split(', ')
```

Comment:
You should append that list of permissions, not rewrite it. It's possible to create multiple filters with the same resource type, but different permissions. In the end, role will receive all the permissions. E.g.:
Filter 1) Resource type: Organization, permissions: view_organizations, create_organizations.
Filter 2) Resource type: Organization, permissions: edit_organizations, destroy_organizations.
Filter 3) Resource type: Organization, permissions: assign_organizations.

Assigning such role will give user sum of all these permissions (thus, user will receive all possible organization permissions).